backend/app/groups/service.py

```
@@ -473,5 +473,45 @@ async def remove_member(self, group_id: str, member_id: str, user_id: str) -> bo
         return result.modified_count == 1
 
+    async def ensure_user_in_group(self, group_id: str, user_id: str) -> dict:
+        """Ensure that the user is a member of the group. Raises HTTPException if not."""
+        db = self.get_db()
+
+        try:
+            obj_id = ObjectId(group_id)
+
+        except errors.InvalidId:
+            logger.warning(f"Invalid group_id: {group_id}")
+            raise HTTPException(status_code=400, detail="Invalid group ID format")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)